parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -585,13 +585,14 @@ configGPUDrivers() {
         exit 1
     fi
 
-    # validate on host, already done inside container.
-    if [[ $OS == $UBUNTU_OS_NAME ]]; then
-        retrycmd_if_failure 120 5 25 nvidia-modprobe -u -c0 || exit $ERR_GPU_DRIVERS_START_FAIL
-    fi
-
+    retrycmd_if_failure 120 5 25 nvidia-modprobe -u -c0 || exit $ERR_GPU_DRIVERS_START_FAIL
```

Comment:
It's already tested by using agentbaker e2e and gpu tests pass

Reply:
The e2e do not fail previously without this change. What happens here is this PR will introduce running the nvidia-ctk command to fix the regression on cgrouv2 k8s environments and that command will fail without this change because certain nvidia modules are not correctly loaded. Thus, we re-added this change to Azure Linux AKS GPU provisioning to correctly load all nvidia modules instead of just relying on installing the Mariner cuda package to do its job.